build/lib/src/builder/build_step.dart

```diff
@@ -80,6 +80,19 @@ abstract class BuildStep implements AssetReader, AssetWriter {
   /// Returns value returned by [action].
   /// [action] can be async function returning [Future].
   T trackStage<T>(String label, T Function() action, {bool isExternal = false});
+
+  /// Indicates that [ids] were read but their content has no impact on the
+  /// outputs of this step.
+  ///
+  /// **WARNING**: Using this introduces serious risk of non-hermetic builds.
+  ///
+  /// If this file changes or becomes unreadable on the next build this build
```

Comment:
Should this be "If any of these files.."?

Reply:
good point, done